src/harness/sas_test_harness.py

```diff
@@ -181,47 +192,58 @@ def __writeDumpFile(self, file_name, data):
+    
     fad_record_list = []
-    generated_file_name = None
-    record_url = None
-    dump_path = self.getDumpFilePath()
-
-    for dump_records in dump_records_list:
-      for index, dump_record in enumerate(dump_records):
-        if dump_record.has_key('id'):
-          if 'cbsd' in dump_record['id']:
-            generated_file_name = '{}.json'.format(dump_record['id'].decode('utf-8').split('/')[-1])
-            record_url = self.getBaseUrl() + '/cbsd/%s' % quote(dump_record['id'])
-          elif 'zone' in dump_record['id']:
-            generated_file_name = '{}.json'.format(dump_record['id']).split('/')[-1]
-            record_url = self.getBaseUrl() + '/' + dump_record['id']
-          elif 'esc_sensor' in dump_record['id']:
-            encoded_sensor_id = str(hashlib.sha1(dump_record['id']).hexdigest()).encode('utf-8')
-            generated_file_name = '{}.json'.format(encoded_sensor_id)
-            record_url = self.getBaseUrl() + '/%s/%s' % (dump_record['id'], encoded_sensor_id)
-          else:
-            raise Exception("ConfigurationError:incorrect record type. Unable to write FAD record")
-        else:
-          raise Exception("ConfigurationError:field id is not found. Unable to write FAD record")
-        if record_url is not None and generated_file_name is not None:
-          self.__writeDumpFile(generated_file_name, dump_record)
-          fad_record = self.__createFadRecord(record_url, dump_record)
-          fad_record_list.append(fad_record)
-        else:
-          raise Exception("SASTestHarnessError:Unable to create record url")
-      fad_file_name = 'FAD.json'
-      full_activity_dump = self.__createFadObject(fad_record_list)
-      self.__writeDumpFile(fad_file_name, full_activity_dump)
-      logging.info("FAD Records generated successfully."
-                   "The files are placed under the path:%s", self.getDumpFilePath())
+    dump_body = {}
+
+    for index, dump_records in enumerate(dump_records_list):
+        if len(dump_records) != 0:
+           dump_body['records'] = dump_records
+           # Get the first element in the dump records. 
+           # First record is used to check type of the dump record
+           dump_record = dump_records[0]
+           if dump_record.has_key('id'):
+              if 'cbsd' in dump_record['id']:
+                 generated_file_name = 'activity_dump_file_cbsd'+str(index)+'.json'
+                 record_url = self.getBaseUrl() +'/cbsd/'+ generated_file_name
+              elif 'zone' in dump_record['id']:
+                 generated_file_name = 'activity_dump_file_zone'+str(index)+'.json'
+                 record_url = self.getBaseUrl() +'/zone/'+ generated_file_name
+              elif 'esc_sensor' in dump_record['id']:
+                 generated_file_name = 'activity_dump_file_esc_sensor'+str(index)+'.json'
+                 record_url = self.getBaseUrl() +'/esc_sensor/'+ generated_file_name 
```

Comment:
Else error. The 'else' below is at the wrong indent level to do this.

Reply:
Corrected indentation